demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp

```diff
@@ -0,0 +1,120 @@
+        ptr[(batch_size + 1) * frame.total()] = cv::Vec3b(0, counter, 0); // position of first and counter
+        /** Time when last image was added to batch **/
+        start_time = std::chrono::steady_clock::now();
+    }
+    virtual bool pull(cv::gapi::wip::Data &data) override {
+        /** Is first already pulled **/
+        if (!first_pulled) {
+            GAPI_Assert(!batch.empty());
+            first_pulled = true;
+            data = batch.clone();
+            return true;
+        }
+        /** Reading of frame with ImagesCapture class if isn't first_pulled **/
+        cv::Mat frame = cap->read();
+        if (!frame.data) {
+            return false;
+        }
+        /** Copy fast/real/current frame from VideoCapture to batch memory as 17th image **/
+        std::copy_n(frame.begin<cv::Vec3b>(),
+                    frame.total(),
+                    batch.begin<cv::Vec3b>() + batch_size * frame.total()); // real frame
+
+        /** Getting of current time **/
+        const auto cur_time = std::chrono::steady_clock::now();
+        /** Counting of gap between last adding of frame to batch **/
+        const auto gap = std::chrono::duration_cast<std::chrono::milliseconds>(cur_time - start_time).count();
+        if (gap > (1.f / batch_fps * 1000)) {
+            /** Update time of last adding of image to batch **/
+            start_time = std::chrono::steady_clock::now();
+            cv::Vec3b* ptr = batch.ptr<cv::Vec3b>();
+            if (counter < batch_size) { // Filling of batch
+                /** Place of new frame in batch **/
+                const auto step = counter * frame.total();
+
+                /** Adding of new image to batch **/
+                std::copy_n(frame.begin<cv::Vec3b>(), frame.total(),
+                            batch.begin<cv::Vec3b>() + step);
+                /** One more image is added **/
+                ++counter;
+
+                /** Putting of info about batch to additional element.
+                 *  First image is 0, counter is 1 - 16(11) **/
+                ptr[(batch_size + 1) * frame.total() + 0] = cv::Vec3b(0, counter, 0);
+            } else if(counter == batch_size) { // batch is filled already rewrite memory of batch
+                /** Adding of new image to batch. Now it is placed on ferst element place. **/
+                std::copy_n(frame.begin<cv::Vec3b>(), frame.total(),
+                            batch.begin<cv::Vec3b>() + first_el * frame.total());
+                /** And first image is next image (second image in the past)**/
+                if (++first_el > (batch_size - 1)) {
+                    first_el = 0; // case when new image is batch_size - 1 (last in batch memory)
+                }
+                /** Putting of info about batch to additional element **/
+                ptr[(batch_size + 1) * frame.total()] = cv::Vec3b(first_el, counter, 1);
+            }
+        }
+        data = batch.clone();
```

Comment:
as I understand idea correctly a network requires 15 frame to make suggestion about movement (initial pose... final pose)
but we are returned from `pull` method every time and 90% duration and it returns not filled `batch`  without 15 filled frames in itself. So in which part we waiting for the last 15th frame right before network starting? does it happen in another thread in custom kernel?

Reply:
and what happens if source is get stucked and cannot  satisfy 15 fps criteria? do we need to reset batch buffer and fill it it from beginning?